### PR TITLE
Add started_elections topic to websockets

### PR DESCRIPTION
**Pull request by gr0vity-dev**
This PR adds a new topic "started_elections" to websockets.

* add observers.active_started for started elections

* Update tests

### CODE REVIEW

File: nano/core_test/websocket.cpp [hunk 1]
```diff
@@ -124,6 +124,46 @@ TEST (websocket, confirmation)
...
 }
 
+// Tests getting notification of a started election
+TEST (websocket, started_election)
+{
+	nano::system system;
+	nano::node_config config (nano::get_available_port (), system.logging);
+	config.websocket_config.enabled = true;
+	config.websocket_config.port = nano::get_available_port ();
+	auto node1 (system.add_node (config));
```
Comment (clemahieu) [inline nano/core_test/websocket.cpp]: We should use the more standard initialisation syntax for new code,

auto node1 = system.add_node (config);

File: nano/core_test/websocket.cpp [hunk 2]
```diff
@@ -124,6 +124,46 @@ TEST (websocket, confirmation)
...
+		EXPECT_EQ (1, node1->websocket_server->subscriber_count (nano::websocket::topic::started_election));
+		return client.get_response ();
+	});
+	auto future = std::async (std::launch::async, task);
+
+	ASSERT_TIMELY (5s, ack_ready);
+
+	// Create election, causing a websocket message to be emitted
+	nano::keypair key1;
+	auto send1 (std::make_shared<nano::send_block> (nano::dev::genesis->hash (), key1.pub, 0, nano::dev::genesis_key.prv, nano::dev::genesis_key.pub, *system.work.generate (nano::dev::genesis->hash ())));
```
Comment (clemahieu) [inline nano/core_test/websocket.cpp]: This should use the block builder to make it easier to understand what the constructor pieces are.

File: nano/core_test/websocket.cpp [hunk 3]
```diff
@@ -124,6 +124,46 @@ TEST (websocket, confirmation)
...
+	nano::publish publish1{ nano::dev::network_params.network, send1 };
+	auto channel1 (node1->network.udp_channels.create (node1->network.endpoint ()));
```
Comment (clemahieu) [inline nano/core_test/websocket.cpp]: auto channel1 = node1->network.udp_channels.create (node1->network.endpoint ());

File: nano/core_test/websocket.cpp [hunk 4]
```diff
@@ -124,6 +124,54 @@ TEST (websocket, confirmation)
...
+	auto send1 = builder
+				 .send ()
+				 .previous (nano::dev::genesis->hash ())
+				 .destination (key1.pub)
+				 .balance (0)
+				 .sign (nano::dev::genesis_key.prv, nano::dev::genesis_key.pub)
+				 .work (*system.work.generate (nano::dev::genesis->hash ()))
+				 .build_shared ();
+	nano::publish publish1{ nano::dev::network_params.network, send1 };
+	auto channel1 = node1->network.udp_channels.create (node1->network.endpoint ());
```
Comment (clemahieu) [inline nano/core_test/websocket.cpp:161]: There's work in progress to remove UDP so this should be using nano::transport::inproc::channel if TCP isn't needed.

Comment (gr0vity-dev) [reply nano/core_test/websocket.cpp:161]: I copied that part from the the stopped_election testcase which also uses udp_channel.

**Review by clemahieu — APPROVED** [review]

**Review by dsiganos — APPROVED** [review]

**Review by pwojcikdev — APPROVED** [review]

**Pull request merged**